src/parsers/manifestjson.js

```diff
@@ -70,6 +70,9 @@ export default class ManifestJSONParser extends JSONParser {
       };
     } else {
       // We've parsed the JSON; now we can validate the manifest.
+      if (this.parsedJSON.browser_specific_settings) {
+        this.parsedJSON.applications = this.parsedJSON.browser_specific_settings;
```

Comment:
@eviljeff Given that on the firefox side the `browser_specific_settings` override the `applications` property only if it also contains the `gecko` property, I think that we should do the same on the linter side.

As an example, if an extension manifest.json have the `browser_specific_settings` property with a sub-property related to another browser, e.g. edge, and also the `applications` property with the 'gecko' property, on the Firefox side we will be using the content of `applications`, and so the linter should possibly do the same.

Reply:
https://github.com/mozilla/addons-linter/pull/2285/commits/93e6b9cfe8f5e044e63cac09253c3953357fe0fc